solver/llbsolver/file/unpack.go

```diff
@@ -0,0 +1,61 @@
+package file
+
+import (
+	"archive/tar"
+	"context"
+	"os"
+	"time"
+
+	"github.com/containerd/continuity/fs"
+	"github.com/docker/docker/pkg/archive"
```

Comment:
can we use containerd/archive? (although not support for chroot yet)

Reply:
Unfortunately not, for example it is missing non-gzip compression support and only works with whiteouts layer stacking mode. I did create a fork https://github.com/tonistiigi/fsutil/tree/archive/archive a while back but didn't feel it was tested/secure enough to be used atm.